### ISSUE TITLE
Enable correlation of multiple resources 

### ISSUE THREAD

**Issue by jonjozwiak** [bridgecrewio/checkov#155]
Several checks such as VPC flow logs, S3 block public access, and CloudFront OAI require correlation of config across multiple resources.  Enable checkov to handle this in a single check

**Comment by jare19** [comment]
VPC flow logs enabled for a newly created VPC is very important to my organization and would love to see this prioritized. Or an idea for a work around would be helpful.

**Comment by christophetd** [comment]
One way to implement it would be to expose a function to custom checks allowing to retrieve resources of a specific type, e.g.

```python
from checkov.common.utils import find_resource_by_type # This does not exist

class MySampleCheck(BaseResourceCheck):
    def __init__(self):
        name = "Custom check"
        id = "SAMPLE"
        supported_resources = ['aws_s3_bucket']
        categories = [CheckCategories.GENERAL_SECURITY]
        super().__init__(name=name, id=id, categories=categories, supported_resources=supported_resources)

    def scan_resource_conf(self, conf):
      bucket_name = conf.get('bucket')[0]
      public_access_blocks = find_resource_by_type("aws_s3_public_access_block")
      for public_access_block in public_access_blocks:
            if public_access_block.get('bucket') == bucket_name:
                return CheckResult.PASSED
      return CheckResult.FAILED
```

**Comment by robeden** [comment]
Flow log request ticket: #154

**Comment by robeden** [comment]
It would be great if we could pass an addition argument in to the `scan_*` functions which would provide access to other parts of the model. That would prevent setting "global" state in the check itself and allow for future possibilities such as async scans (not that this is an imminent thing, but good to allow for it). I think this would be the cleanest and most intuitive place to put an accessor object so that devs know it exists.

However, assuming we don't want to change the function signature, the way to go about this would be to set a field in the base check during the duration of the `run` function:
```
def run(...):
   self.doc_access = ...
   try:
      ...
   finally:
      self.doc_access = None
```

Then there could be support functions such as the one @christophetd describes above.

**Comment by robeden** [comment]
I'm working on this...

**Comment by robeden** [comment]
Fixed by #1023
